### PR TITLE
Remove fixme comment about clobber_abi on PowerPC

### PR DESCRIPTION
**Pull request by taiki-e**
This was considered an unresolved question in https://github.com/rust-lang/rust/pull/131341, but according to the ABI document published in 2011 by Power.org the current implementation is fine as-is.
https://github.com/rust-lang/rust/pull/131341#discussion_r1829358396

> According to [Power Architecture 32-bit Application Binary Interface Supplement 1.0 - Linux & Embedded](https://web.archive.org/web/20120608163804/https://www.power.org/resources/downloads/Power-Arch-32-bit-ABI-supp-1.0-Unified.pdf) published in 2011, PPC32 has the same convention here as PPC64.
> 
> Therefore, we can just remove the FIXME comment here.

r? workingjubilee

### CODE REVIEW

File: compiler/rustc_target/src/asm/mod.rs [hunk 1]
```diff
@@ -1138,8 +1138,6 @@ impl InlineAsmClobberAbi {
                     f8, f9, f10, f11, f12, f13,
 
                     // v0-v19
-                    // FIXME: PPC32 SysV ABI does not mention vector registers processing.
-                    // https://refspecs.linuxfoundation.org/elf/elfspec_ppc.pdf
```
Comment (jieyouxu) [inline compiler/rustc_target/src/asm/mod.rs:1142]: Suggestion: can you replace this with the link in the PR description? Can't hurt to have some useful links here :3

Comment (taiki-e) [reply compiler/rustc_target/src/asm/mod.rs:1142]: Added links (there are six links).

https://github.com/rust-lang/rust/blob/e2d50315efb9bcf5ad73bd4cb77f3d8f72a7d26a/compiler/rustc_target/src/asm/mod.rs#L1131-L1144

Comment (jieyouxu) [reply compiler/rustc_target/src/asm/mod.rs:1142]: Yes, that's very nice, thank you!

**Comment by jieyouxu** [comment]
Thanks!

@bors r+ rollup

**Pull request merged**